packages/cadl-ranch-specs/http/azure/resource-manager/models/resources/mockapi.ts

```diff
@@ -42,6 +42,162 @@ const validNestedResource = {
   },
 };
 
+const validExtensionResource = {
+  id: `/subscriptions/${SUBSCRIPTION_ID_EXPECTED}/resourceGroups/${RESOURCE_GROUP_EXPECTED}/providers/Azure.ResourceManager.Models.Resources/extensionTrackedResources/extension`,
+  name: "roleAssign",
```

Comment:
`name` property value(`roleAssign`) and the one in the path(`extension`) don't match. Maybe just `extension`.